lib/hashtable.cc

```diff
@@ -997,4 +997,79 @@ void Hashtable::get_kmer_counts(const std::string &s,
     }
 }
 
+void Hashtable::find_high_degree_nodes(const char * s,
+                                       SeenSet& high_degree_nodes)
+    const
+{
+    Traverser traverser(this);
+    KmerIterator kmers(s, _ksize);
+
+    unsigned long n = 0;
+    while(!kmers.done()) {
+        n++;
+        if (n % 10000 == 0) {
+            std::cout << "... find_high_degree_nodes: " << n << "\n";
+            std::cout << std::flush;
+        }
+        Kmer kmer = kmers.next();
+        if ((traverser.degree(kmer)) > 2) {
+            high_degree_nodes.insert(kmer);
+        }
+    }
+}
+
+unsigned int Hashtable::traverse_linear_path(const Kmer seed_kmer,
+                                             SeenSet &adjacencies,
+                                             SeenSet &visited, Hashtable &bf,
+                                             SeenSet &high_degree_nodes)
+    const
+{
+    unsigned int size = 0;
+
+    auto filter = [&] (Kmer& n) -> bool {
+        return true;
+    };
```

Comment:
With my primarily ANSI C background I'm not sure how to parse this. Pointers? (no pun intended)